### PR TITLE
util/timeutil: migrate test-infra to testify

### LINKED ISSUE #26426: migrate test-infra to testify for `util/timeutil` pkg

### PR DESCRIPTION
**Pull request by top628LJ**
Signed-off-by: top628LJ <928073183@qq.com>

### What problem does this PR solve?

Issue Number: close #26426 

### What is changed and how it works?

What's Changed: migrate test-infra to testify for util/timeutil
 
### Release note 
- No release note

### CODE REVIEW

File: util/timeutil/time_test.go [hunk 1]
```diff
@@ -23,67 +23,65 @@ import (
...
-}
-
-type testTimeSuite struct{}
-
-func (s *testTimeSuite) TestgetTZNameFromFileName(c *C) {
 	tz, err := inferTZNameFromFileName("/usr/share/zoneinfo/Asia/Shanghai")
 
-	c.Assert(err, IsNil)
-	c.Assert(tz, Equals, "Asia/Shanghai")
+	assert.Nil(t, err)
```
Comment (tisonkun) [inline util/timeutil/time_test.go:34]:
```suggestion
	require.NoError(t, err)
```

You may use more precious assertions. Also `require` plays the same as `c.Assert` which `FailNow` if assertion failed.

**Review by tisonkun — COMMENTED** [review]
Generally looks good. Comments inline.

File: util/timeutil/time_test.go (hunk 1 above)
Comment (tisonkun) [reply util/timeutil/time_test.go:34]: ditto others.

Comment (top628LJ) [reply util/timeutil/time_test.go]: got it and updated

**Review by xhebox — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]
LGTM

**Review by disksing — APPROVED** [review]

**Pull request merged**